awx/main/dispatch/worker/callback.py

```diff
@@ -98,7 +98,8 @@ def flush(self, force=False):
                                 # work around - if the host disappears before
                                 # the event can be processed
                                 e.host_id = None
-                            e.save()
+                            if not e.skip_save():
```

Comment:
I'm not certain this is going to work.  This code path is the *fallback* in case the `bulk_create` above fails (which almost never happens in practice).